tantar/src/main/java/org/opencadc/tantar/Main.java

```diff
@@ -0,0 +1,183 @@
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.tantar;
+
+import ca.nrc.cadc.auth.SSLUtil;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.StringUtil;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Properties;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.storage.StorageAdapter;
+import org.opencadc.tantar.policy.ResolutionPolicy;
+
+
+/**
+ * Main application entry.  This class expects a tantar.properties file to be available and readable.
+ */
+public class Main {
+
+    private static final String CONFIG_KEY_PREFIX = "org.opencadc.tantar";
+
+    private static final String BUCKET_KEY = String.format("%s.bucket", CONFIG_KEY_PREFIX);
+    private static final String CONFIGURATION_DIRECTORY = System.getProperty("DEFAULT_CONFIG_DIR", "/config");
+    private static final String CERTIFICATE_FILE_LOCATION = String.format("%s/cadcproxy.pem", CONFIGURATION_DIRECTORY);
+    private static final String CONFIGURATION_FILE_LOCATION = String.format("%s/tantar.properties",
+                                                                            CONFIGURATION_DIRECTORY);
+    private static final String REPORT_ONLY_KEY = String.format("%s.reportOnly", CONFIG_KEY_PREFIX);
+
+    private static final Logger LOGGER = Logger.getLogger(Main.class);
+    private static final Reporter REPORTER = new Reporter(LOGGER);
+
+    public static void main(final String[] args) {
+        Main.REPORTER.start();
+        Main.configure();
+        Main.setLogging();
+
+        try {
+            final StorageAdapter storageAdapter = InventoryUtil.loadPlugin(StorageAdapter.class);
+
+            final String bucket = System.getProperty(BUCKET_KEY);
+            if (!StringUtil.hasLength(bucket)) {
+                throw new IllegalArgumentException(String.format("Bucket is mandatory.  Please set the %s property.",
+                                                                 BUCKET_KEY));
+            }
+
+            final boolean reportOnlyFlag = Boolean.parseBoolean(System.getProperty(REPORT_ONLY_KEY,
+                                                                                   Boolean.FALSE.toString()));
+
+            if (reportOnlyFlag) {
+                LOGGER.info("*********");
+                LOGGER.info("********* Reporting actions only.  No actions will be taken. *********");
+                LOGGER.info("*********");
+            }
+
+            final ResolutionPolicy resolutionPolicy = InventoryUtil.loadPlugin(ResolutionPolicy.class, Main.REPORTER,
+                                                                               reportOnlyFlag);
+
```

Comment:
I would still prefer that reportOnly be passed to the BucketValidator rather than relying on all policy implementations to do it correctly. BV can easily just do a no-op return. 

And it still looks like inverting the sequence here (create the BV fist and pass it to the ctor of the ResolutionPolicy) is more obvious since the RP needs a ValidateEventListener to call and currently relies on subscribe() being called exactly once. I'm not against listener pattern, but it only really makes sense if the problem requires 0..n listeners and not exactly one.